src/analysis/processing/qgsalgorithmpointstopaths.cpp

```diff
@@ -62,37 +62,39 @@ QString QgsPointsToPathsAlgorithm::groupId() const
 
 void QgsPointsToPathsAlgorithm::initAlgorithm( const QVariantMap & )
 {
-  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+  addParameter( std::make_unique<QgsProcessingParameterFeatureSource>( QStringLiteral( "INPUT" ),
                 QObject::tr( "Input layer" ), QList< int >() << static_cast< int >( Qgis::ProcessingSourceType::VectorPoint ) ) );
-  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "CLOSE_PATH" ),
+  addParameter( std::make_unique<QgsProcessingParameterBoolean>( QStringLiteral( "CLOSE_PATH" ),
                 QObject::tr( "Create closed paths" ), false, true ) );
-  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "ORDER_EXPRESSION" ),
+  addParameter( std::make_unique<QgsProcessingParameterExpression>( QStringLiteral( "ORDER_EXPRESSION" ),
                 QObject::tr( "Order expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
-  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "NATURAL_SORT" ),
+  addParameter( std::make_unique<QgsProcessingParameterBoolean>( QStringLiteral( "NATURAL_SORT" ),
                 QObject::tr( "Sort text containing numbers naturally" ), false, true ) );
-  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "GROUP_EXPRESSION" ),
+  addParameter( std::make_unique<QgsProcessingParameterExpression>( QStringLiteral( "GROUP_EXPRESSION" ),
                 QObject::tr( "Path group expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
-  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ),
+  addParameter( std::make_unique<QgsProcessingParameterFeatureSink>( QStringLiteral( "OUTPUT" ),
                 QObject::tr( "Paths" ), Qgis::ProcessingSourceType::VectorLine ) );
   // TODO QGIS 4: remove parameter. move logic to separate algorithm if needed.
-  addParameter( new QgsProcessingParameterFolderDestination( QStringLiteral( "OUTPUT_TEXT_DIR" ),
+  addParameter( std::make_unique<QgsProcessingParameterFolderDestination>( QStringLiteral( "OUTPUT_TEXT_DIR" ),
                 QObject::tr( "Directory for text output" ), QVariant(), true, false ) );
-  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "NUM_PATHS" ), QObject::tr( "Number of paths" ) ) );
+  addOutput( std::make_unique<QgsProcessingOutputNumber>( QStringLiteral( "NUM_PATHS" ), QObject::tr( "Number of paths" ) ) );
 
   // backwards compatibility parameters
   // TODO QGIS 4: remove compatibility parameters and their logic
-  QgsProcessingParameterField *orderField = new QgsProcessingParameterField( QStringLiteral( "ORDER_FIELD" ),
-      QObject::tr( "Order field" ), QVariant(), QString(), Qgis::ProcessingFieldParameterDataType::Any, false, true );
+  auto orderField = std::make_unique<QgsProcessingParameterField>( QStringLiteral( "ORDER_FIELD" ),
```

Comment:
> I don't think we should use auto with these plain and not-that-long type.

oh please please, can't we make an exception here...? I dislike so much reading QGIS code because of all that type boilerplate that brings more noise than value. Here the scope of that variable is so reduced (3 lignes), so that ``àuto`` doesn't hurt. (in my wishlist I would like to kill all those painful explicit QStringLiteral() or QLatin1String() stuff that also make the code hard to read. In an ideal world, the compiler would figure that out itself)

Reply:
I think we need to explicitly describe when we can use auto or not. I was thinking about this [comment](https://github.com/qgis/QGIS/pull/59259#discussion_r1847636668) when I made mine.

But in the end, that's a different scenario. Just a quicklook on [stackoverflow](https://stackoverflow.com/questions/6900459/the-new-keyword-auto-when-should-it-be-used-to-declare-a-variable-type) and it looks new/make_unique instanciation are also good place to use auto.

So ok then :)

Reply:
I'm ok with auto being used as the type for a std::make_unique call, since it's still immediately clear what the type is when reviewing code. 

(That's my big issue with auto -- it makes in-depth reviews **impossible** to do without first loading the code into an IDE.)

I suggest we need a QEP for this policy, and we could explicitly document where auto should be used vs not.

Reply:
baby steps towards having a place to describe coding policies: https://github.com/qgis/QGIS-Enhancement-Proposals/pull/310